### PR TITLE
[Multi-User Part 7]: Improve Sign-In UX & Rename DB Models for Readability

### PR DESCRIPTION
**Pull request by debanjum**
### ✨ New
- f585a71 Create profile pic drop-down menu in navigation pane
  Put settings page, logout action under drop-down menu

### ⚙️ Fix
- f77336ba Add Key icon for API keys table on Web Client's settings page

### 🧪 Improve
- 87e6b1ea Rename `TextEmbeddings` to `TextEntries` for improved readability
- bcbee05a Rename `Db.Models` `Embeddings`, `EmbeddingsAdapter` to `Entry`, `EntryAdapter`
- 58a71719 Show truncated API key for identification & restrict table width for config page responsiveness

### Screenshots

| Before | After
| --- | --- |
|   |  |

### CODE REVIEW

File: src/khoj/routers/web_client.py [hunk 1]
```diff
@@ -34,19 +34,52 @@
...
+        },
+    )
 
 
 @web_client.get("/chat", response_class=FileResponse)
 @requires(["authenticated"], redirect="login_page")
 def chat_page(request: Request):
-    return templates.TemplateResponse("chat.html", context={"request": request, "demo": state.demo})
+    user = request.user.object
+    user_picture = request.session.get("user", {}).get("picture")
```
Comment (sabaimran) [inline src/khoj/routers/web_client.py:72]: Is the `picture` field always populated? What happens if the user hasn't set a profile picture? Can we supply a generic/default avatar?

File: src/khoj/interface/web/chat.html [hunk 2]
```diff
@@ -281,10 +282,17 @@
                 </a>
             {% endif %}
             <nav class="khoj-nav">
-                <a class="khoj-nav khoj-nav-selected" href="/chat">Chat</a>
-                <a class="khoj-nav" href="/">Search</a>
+                <a class="khoj-nav khoj-nav-selected" href="/chat">💬 Chat</a>
+                <a class="khoj-nav" href="/">🔎 Search</a>
                 {% if not demo %}
-                    <a class="khoj-nav" href="/config">Settings</a>
+                    <!-- Dropdown Menu -->
```
Comment (sabaimran) [inline src/khoj/interface/web/chat.html:288]: Could it be worthwhile to use Jinja Macros to define this account component? That way we won't have to copy/paste as much and we'll get better re-use. https://jinja.palletsprojects.com/en/3.0.x/templates/#macros

File: src/khoj/processor/text_to_jsonl.py [hunk 3]
```diff
@@ -12,14 +12,14 @@
 from khoj.utils.rawconfig import Entry
```
Comment (sabaimran) [inline src/khoj/processor/text_to_jsonl.py:12]: All these `x_to_jsonl` files also need to be renamed to something like `x_to_entries` or `x_to_embeddings`. Do you want to take care of that as part of this PR as well?

File: src/khoj/routers/web_client.py [hunk 4]
```diff
@@ -34,19 +34,52 @@
...
-    return templates.TemplateResponse("index.html", context={"request": request, "demo": state.demo})
+    user = request.user.object
+    user_picture = request.session.get("user", {}).get("picture")
+
+    return templates.TemplateResponse(
+        "index.html",
+        context={
+            "request": request,
+            "demo": state.demo,
+            "username": user.username if user else None,
```
Comment (sabaimran) [inline src/khoj/routers/web_client.py:45]: `user` should never be `None` here. Same with the `index_post` method below.
```suggestion
            "username": user.username,
```

**Comment by sabaimran** [comment]
Can you also include a screenshot? 😁

**Comment by debanjum** [comment]
> Can you also include a screenshot? 😁

Added before, after screenshot

File: src/khoj/routers/web_client.py (hunk 1 above)
Comment (debanjum) [reply src/khoj/routers/web_client.py:72]: Updated code to use username initial as default profile picture. This matches behavior in other applications for default profile picture

File: src/khoj/processor/text_to_jsonl.py (hunk 3 above)
Comment (debanjum) [reply src/khoj/processor/text_to_jsonl.py:12]: Good idea. Updated `s/x_to_jsonl/x_to_entries` for files and `s/XToJsonl/XToEntries/` for class names

File: src/khoj/routers/web_client.py (hunk 4 above)
Comment (debanjum) [reply src/khoj/routers/web_client.py]: Valid point. Fixed

File: src/khoj/interface/web/chat.html (hunk 2 above)
Comment (debanjum) [reply src/khoj/interface/web/chat.html]: Great idea, didn't know that existed. Reduced duplication of HTML by creating a `utils.html` with the macro to generate the navigation pane

**Pull request merged**